### PR TITLE
[FFI] Handling the default library loading on z/OS

### PR DESCRIPTION
**Pull request by ChengJin01**
The changes follow the same way of accessing the default
libraries on AIX with the inlined functions in a list to
address the issue with the default libraries during the
FFI downcall on z/OS.

Related: [Internal 441](https://github.ibm.com/runtimes/javanext/issues/441)

Signed-off-by: ChengJin01 <jincheng@ca.ibm.com>

### CODE REVIEW

**Comment by ChengJin01** [comment]
The PR (verified on AIX) has the same changes as https://github.com/ibmruntimes/openj9-openjdk-jdk21/pull/159.

**Comment by ChengJin01** [comment]
Reviewer: @tajila 
FYI: @keithc-ca, @pshipton

File: src/java.base/share/classes/jdk/internal/foreign/SystemLookup.java [hunk 1]
```diff
@@ -82,16 +87,38 @@ private static SymbolLookup makeSystemLookup() {
...
 
+    /* Obtain the address of the inlined function list by loading the specified library. */
+    private static MemorySegment getInlinedFunctListAddr() {
+        /* Directly load the specified library only once with the inlined libc functions
+         * missing in the default library.
+         */
+        SymbolLookup funcsLibLookup = libLookup(libs -> libs.load(jdkLibraryPath("syslookup")));
+
+        /* Perform a FFI downcall to obtain the address of the inlined function list. */
+        MemorySegment funcs_addr = funcsLibLookup.findOrThrow("funcs_addr");
```
Comment (keithc-ca) [inline src/java.base/share/classes/jdk/internal/foreign/SystemLookup.java:98]: `findOrThrow()` is only available in the head stream. Please revert to:
```java
        MemorySegment funcs_addr = funcsLibLookup.find("funcs_addr").orElseThrow();
```

Comment (ChengJin01) [reply src/java.base/share/classes/jdk/internal/foreign/SystemLookup.java]: Just notice and reverted.

**Comment by keithc-ca** [comment]
Jenkins test sanity aix,win jdk22

**Comment by ChengJin01** [comment]
The failure at https://openj9-jenkins.osuosl.org/job/Test_openjdk22_j9_sanity.functional_x86-64_windows_Personal/6/artifact/Test_openjdk22_j9_sanity.functional_x86-64_windows_Personal_testList_0/5/aqa-tests/TKG/output_17168495324634/Test_openjdk22_j9_sanity.functional_x86-64_windows_Personal_testList_0.tap is the same as explained at https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/794#issuecomment-2134332526 which we should skip `getInlinedFunctListAddr()` on non-AIX/zOS platforms.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**